ping-thing-client.mjs

```diff
@@ -119,39 +114,30 @@ async function pingThing() {
 
         console.log(`Sending ${signature}`);
 
-        const abortController = new AbortController();
-
         while (true) {
           try {
             [slotSent] = await Promise.all([
               getNextSlot(),
-              mSendTransaction(transactionSignedWithFeePayer, {
-                commitment: "confirmed",
-                maxRetries: 0n,
-                skipPreflight: true,
-              }),
-            ]);
-
-            await Promise.race([
-              getRecentSignatureConfirmationPromise({
-                signature,
-                commitment: "confirmed",
-                abortSignal: abortController.signal,
-              }),
-              sleep(TX_RETRY_INTERVAL * txSendAttempts).then(() => {
-                throw new Error("Tx Send Timeout");
-              }),
+              safeRace([
+                mSendAndConfirmTransaction(transactionSignedWithFeePayer, {
+                  commitment: "confirmed",
+                }),
+                timeout(TX_RETRY_INTERVAL * txSendAttempts),
```

Comment:
You can still achieve the timeout behavior by racing the all-up send-and-confirmer with a timeout promise.